server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -283,6 +323,18 @@ export class MongoCollection<T> implements core.ICollection<T>, core.IRetryable
 
         return properties;
     }
+
+    private sanitizeError(error: any) {
+        if (error) {
+            Object.keys(error).forEach((key) => {
+                if (!errorResponseKeysAllowList.includes(key)) {
+                    error[key] = errorSanitizationMessage;
+                } else if (typeof error[key] === "object") {
+                    this.sanitizeError(error[key]);
```

Comment:
What if an error is like:
```
{
     code: '123',
     innerException: this
}
```

I think JS have the problem of [circular reference](https://stackoverflow.com/a/1493501)

We probably need to check if there are libraries similar json exception library

Reply:
I think normally we should not expect a circular reference in Mongo error response. Both the data schema and Mongo error response schema do not have/assume circular references. Can't think of a way a circular reference will be in error response.